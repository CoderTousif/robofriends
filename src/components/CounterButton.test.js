import React from 'react';
import CounterButton from './CounterButton';
import { shallow } from 'enzyme';

it('should render counterbutton component', () => {
    expect(shallow(<CounterButton />)).toMatchSnapshot();
});

it('should render counterbutton component correctly', () => {
    const wrapper = shallow(<CounterButton />);

    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');

    expect(wrapper.state()).toEqual({ count: 2 });

    wrapper.find('[id="counter"]').simulate('keypress');

    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual('primary');
});
