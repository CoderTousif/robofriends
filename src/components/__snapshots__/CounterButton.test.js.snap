// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render counterbutton component 1`] = `ShallowWrapper {}`;
